pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -1369,3 +1372,58 @@ def bubble_sort(array, **kwargs):
         array._modify(force=True)
 
     return array
+
+def linear_search(array, elem):
+    for i in range(array._size):
```

Comment:
Is `array.size` not working? We shouldn't access private attributes (the ones having `_` as prefix) unless really needed.

Reply:
no no. While seeing the code, I noticed that only _size is being used, so I used that too